services/spar/test-integration/Test/Spar/APISpec.hs

```diff
@@ -618,14 +618,14 @@ specCRUDIdentityProvider = do
           callIdpGet' (env ^. teSpar) (Just userid) idpid
             `shouldRespondWith` checkErr (== 404) "not-found"
 
-        context "with email" $ it "responds with 2xx and removes IdP" $ do
+        context "with email, idp non-empty" $ it "responds with 412 and does not remove IdP" $ do
           env <- ask
           (firstOwner, tid, idp) <- registerTestIdP
           ssoOwner <- mkSsoOwner firstOwner tid idp
           callIdpDelete' (env ^. teSpar) (Just ssoOwner) (idp ^. idpId)
-            `shouldRespondWith` \resp -> statusCode resp < 300
+            `shouldRespondWith` checkErr (== 412) "idp-has-bound-users"
```

Comment:
these error code changes confuse me, as i don't see 404 or 412 in the changed code.

Reply:
> these error code changes confuse me, as i don't see 404 or 412 in the changed code.

In [this line](https://github.com/wireapp/wire-server/blob/373063aa4f7dc42073dab5a4a766ab7e412a6c65/services/spar/src/Spar/API.hs#L182), we change the behavior that we need to align these tests with: before this change, deleting IdPs populated with wire users was fine; now [it's not](zinfra/backend-issues#754)).